islandora_xmlsitemap.install

```diff
@@ -13,7 +13,83 @@ function islandora_xmlsitemap_uninstall() {
     'islandora_xmlsitemap_number_of_pids_to_process',
     'islandora_xmlsitemap_last_modified_value',
     'islandora_xmlsitemap_last_modified_field',
-    'islandora_namespace_restriction_enforced',
   );
   array_walk($variables, 'variable_del');
 }
+
+/**
+ * Implements hook_install().
+ */
+function islandora_xmlsitemap_install() {
+  // XXX: Taking advantage of the update hook's batch functionality to run the
+  // same batch on install as one would have to in the update hook.
+  islandora_xmlsitemap_update_7000();
+}
+
+/**
+ * Implements hook_schema().
+ */
+function islandora_xmlsitemap_schema() {
+  return array(
+    'islandora_xmlsitemap_entry_ids' => array(
+      'description' => 'Autoincrementing table to store IDs we can use for xmlsitemap table entries.',
+      'fields' => array(
+        'id' => array(
+          'description' => 'Current ID number',
+          'type' => 'serial',
+          'not null' => TRUE,
+        ),
+        'pid' => array(
+          'description' => 'The PID relating to this ID.',
+          'type' => 'varchar',
+          'length' => 255,
+        ),
+      ),
+      'primary key' => array('id'),
+    ),
+  );
+}
+
+/**
+ * Add the ID table schema in.
+ *
+ * XXX: satisfies a race condition (details in the @see below); the best way to
+ * satisfy this condition was using an auto-incrementing ID in a table, but if
+ * islandora_xmlsitemap is already in use, the ID likely can't simply start at
+ * 1. This function iterates the auto-incrementing ID field to the point where
+ * it's supposed to be. It's a little bit goofy to do it this way, but it
+ * ensures that the method we're using to set the initial ID is database-
+ * agnostic.
+ *
+ * @see https://jira.duraspace.org/browse/ISLANDORA-1784
+ */
+function islandora_xmlsitemap_update_7000(&$sandbox = array()) {
+  module_load_include('inc', 'islandora_xmlsitemap', 'includes/db');
+  if (!isset($sandbox['target'])) {
+    // Create our table, if it doesn't exist.
+    if (!db_table_exists('islandora_xmlsitemap_entry_ids')) {
+      $schema = islandora_xmlsitemap_schema();
+      db_create_table('islandora_xmlsitemap_entry_ids', $schema['islandora_xmlsitemap_entry_ids']);
+    }
+    // Set an initial value.
+    islandora_xmlsitemap_insert_increment_record(NULL);
+    // And immediately give it the boot.
+    islandora_xmlsitemap_delete_increment_record(NULL);
+    // Get the current max.
+    $q = db_select('xmlsitemap', 'x');
+    $q->addExpression('MAX(id)');
+    $q->condition('type', 'custom', '=');
+    $current_id = $q->execute()->fetchField();
+    $sandbox['target'] = $current_id !== FALSE ? $current_id : 1;
+    $sandbox['current'] = 1;
+  }
+
+  // Start filling in records.
+  foreach (array_fill(0, min(1000, $sandbox['target'] - $sandbox['current']), NULL) as $increment) {
```

Comment:
I think we have an off by one error here. Imagine there are 1001 items.  I think this would think it had finished.